node/as/thrift.js

```diff
@@ -119,6 +119,10 @@ TChannelAsThrift.prototype.request = function request(reqOptions) {
     var self = this;
 
     assert(self.channel, 'channel is required for thrift.request()');
+    assert(reqOptions &&
+        reqOptions.type !== 'tchannel.request' &&
```

Comment:
why is `reqOptions.type` even a thing? what are you protecting against that we saw here? it seems to me like we're simply trying to say "does it look like they passed a request object in instead of an options object"... and when is "type" ever a valid option?